go/pkg/retry/retry.go

```diff
@@ -60,6 +61,27 @@ const UnlimitedAttempts = Attempts(0)
 // Always always retries, regardless of error.
 func Always(error) bool { return true }
 
+// TransientOnly returns true if the error is transient.
+// It implements ShouldRetry type.
+func TransientOnly(err error) bool {
+	// Retry RPC timeouts. Note that we do *not* retry context cancellations (context.Cancelled);
+	// if the user wants to back out of the call we should let them.
+	if err == context.DeadlineExceeded {
```

Comment:
better to use https://golang.org/pkg/errors/#Is ?

Reply:
yeah, this code should probably use this func, but this func is a part of "errors" package, while this repo seems to be using `github.com/pkg/errors` (i.e. it uses `errors.Wrapf` instead of `fmt.Errorf("foobar: %w", err)`).

Maybe this repo should switch from `github.com/pkg/errors` to `errors`, but this seems out of scope, and probably warrants a discussion with the team.

Reply:
I think it is fine to use the function as `github.com/pkg/errors` supports that like
https://github.com/pkg/errors/blob/614d223910a179a466c1767a985424175c39b465/go113.go#L16

Reply:
sorry, i meant this file is already using a different `errors` package.
But sure, I've added `errors.Is`, with a package alias. Looks awkward though. Hopefully it is temporary.

Reply:
Ah, I see.

Reply:
FWIW I like `errors.Is`. We should converge the errors package to fmt.Errorf and errors.Is. Realistically, we're probably not going to touch it much :)

Reply:
Turns out `github.com/pkg/errors` also has `Is`: https://pkg.go.dev/github.com/pkg/errors#Is
maybe we need to update our pin.